### PR TITLE
Use await instead of promises

### PR DESCRIPTION
**Pull request by fhinkel**
Drive-by fix: Run linter for template strings.

### CODE REVIEW

**Comment by ace-n** [comment]
This _may_ have been obviated by #1489 🙂

**Comment by fhinkel** [comment]
Haha, yes, most is obviated. But a few last changes are left.

File: functions/slack/index.js [hunk 1]
```diff
@@ -106,26 +106,23 @@ const verifyWebhook = body => {
...
-        console.log(err);
-        if (err) {
-          reject(err);
-          return;
-        }
-
-        // Return a formatted message
-        resolve(formatSlackMessage(query, response));
+const makeSearchRequest = async query => {
+  await kgsearch.entities.search(
```
Comment (ace-n) [inline functions/slack/index.js]: 1) Do we know if this _actually_ returns a promise? (either via running it ourselves, or reviewing Slack's docs)
2) If so, we should get rid of the callback below. 🙂

File: functions/slack/index.js [hunk 2]
```diff
@@ -161,12 +158,10 @@ exports.kgSearch = async (req, res) => {
 
     // Send the formatted message back to Slack
     res.json(response);
-
-    return Promise.resolve();
   } catch (err) {
     console.error(err);
     res.status(err.code || 500).send(err);
-    return Promise.reject(err);
+    throw err;
```
Comment (ace-n) [inline functions/slack/index.js]: Nit (perhaps better addressed in a different PR): redundant?

File: functions/spanner/index.js [hunk 3]
```diff
@@ -50,9 +50,7 @@ exports.get = async (req, res) => {
     const rows = results[0].map(row => row.toJSON());
     rows.forEach(row => {
       res.write(
-        `SingerId: ${row.SingerId}, ` +
-          `AlbumId: ${row.AlbumId}, ` +
-          `AlbumTitle: ${row.AlbumTitle}\n`
+        `SingerId: ${row.SingerId}, AlbumId: ${row.AlbumId}, AlbumTitle: ${row.AlbumTitle}\n`
```
Comment (ace-n) [inline functions/spanner/index.js]: I believe @grant [was 👎on this](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/1489#discussion_r326799551).

File: functions/tokenservice/functions/index.js [hunk 4]
```diff
@@ -149,21 +149,17 @@ exports.getOAuthToken = functions.https.onCall(async (data, context) => {
...
+    const doc = await docRef.get();
     if (doc.exists && isValid(doc.data().expireTime)) {
-      //push notification
       pushNotification(
         data['deviceID'],
         doc.data().accessToken,
         doc.data().expireTime
       );
-      return doc.data();
+      await doc.data();
```
Comment (ace-n) [inline functions/tokenservice/functions/index.js]: This seems confusing - why are we using `doc.data()` presumably _before_ we've `await`ed it? (If we know this case _definitely won't happen_, we should at least leave a comment explaining that.)

cc @santhoshvaddi

File: functions/tokenservice/functions/index.js [hunk 5]
```diff
@@ -149,21 +149,17 @@ exports.getOAuthToken = functions.https.onCall(async (data, context) => {
...
     } else {
       const result = await retrieveCredentials(context);
-      console.log('Print result from retrieveCredentials functions');
```
Comment (ace-n) [inline functions/tokenservice/functions/index.js]: @santhoshvaddi I assume this deletion is fine with you?

File: functions/tokenservice/functions/index.js [hunk 6]
```diff
@@ -175,7 +171,7 @@ exports.getOAuthToken = functions.https.onCall(async (data, context) => {
     console.log('Error retrieving token', err);
     pushNotification(data['deviceID'], 'Error retrieving token', 'Error');
     // return 'Error retrieving token';
```
Comment (ace-n) [inline functions/tokenservice/functions/index.js]: Nit: redundant comment, remove?

File: functions/tokenservice/functions/index.js (hunk 4 above)
Comment (santhoshvaddi) [reply functions/tokenservice/functions/index.js]: After push notification we just want to return from the function. The client will receive the token in the form of push notification. I don't think `await` is required here.

File: functions/tokenservice/functions/index.js (hunk 5 above)
Comment (santhoshvaddi) [reply functions/tokenservice/functions/index.js]: Yes, this is fine with me.

File: functions/tokenservice/functions/index.js (hunk 6 above)
Comment (santhoshvaddi) [reply functions/tokenservice/functions/index.js]: Yes, we can remove this.

**Pull request closed without merging**